dashboard/lib/api/v1/school_autocomplete.rb

```diff
@@ -13,9 +13,12 @@ class Api::V1::SchoolAutocomplete
   # @return [Array] an array of JSON formatted schools
   def get_matches(query, limit)
     return [] if query.length < MIN_QUERY_LENGTH
-    schools = School.
-      where("MATCH(name,city) AGAINST(? IN BOOLEAN MODE)", to_search_string(query)).
-      limit(limit)
+    schools = School.limit(limit)
+    if query.match(/^\d{,5}$/).nil?
```

Comment:
What about extended zips (5-4 format)? Should we be reusing this [regex](https://github.com/code-dot-org/code-dot-org/blob/d1a1886c5651547d03082e8ac7a2ea72e3e301db/lib/cdo/regexp.rb#L19)? This could be `unless RegexpUtils.us_zip_code?(query)` if it's meant to match the entire string.

Also note there's a nice regex match shorthand, [=~](https://ruby-doc.org/core-2.2.0/Regexp.html#method-i-3D-7E), especially (as with this case) you're not using the resulting match object.

Instead of `if query.match(regex).nil?`, you can do `unless query =~ regex`. Though rubocop will complain about unless / else, so you'll want to reverse the clauses and use `if`.

Reply:
We don't store extended ZIPs currently in schools. If we wanted to support that, then I think the correct path would be to truncate anything after "-".

Also, it's possible that the ZIP is not fully specified when the request is made, so we may get only four digits, in which case we would return some initial results.

I found the `=~` operator results seem to be confusing at first which is why I opted for the `match` method and the `nil?`. It matches the regular expression against a string, and it returns either the offset of the match from the string if it is found, otherwise nil.

Reply:
re: `=~` being confusing, in this case we're not using the resulting value, we just want to know whether or not it found a match (whether it's falsy or truthy) so I'd argue it's less confusing. 

On the other hand this `.nil?` at the end of the line is easy to miss. If you'd prefer sticking with `match` over `=~` then I'd still suggest switching to `unless query.match(...)` (or better `if query.match(...)` and swapping the clauses) to this `if query.match(...).nil?` pattern.

Reply:
Updated, and moved the logic to another method.